ui/app/pages/connected-sites/connected-sites.container.js

```diff
@@ -47,17 +64,38 @@ const mapDispatchToProps = (dispatch) => {
 }
 
 const mergeProps = (stateProps, dispatchProps, ownProps) => {
-  const { domains, selectedAddress, tabToConnect } = stateProps
+  const {
+    connectedDomains,
+    domains,
+    selectedAddress,
+    tabToConnect,
+  } = stateProps
   const {
     disconnectAccount,
+    disconnectAllAccounts,
     legacyExposeAccounts: dispatchLegacyExposeAccounts,
   } = dispatchProps
+  const { history } = ownProps
+
+  const closePopover = () => history.push(DEFAULT_ROUTE)
 
   return {
     ...ownProps,
     ...stateProps,
     ...dispatchProps,
-    disconnectAccount: (domainKey) => disconnectAccount(domainKey, domains[domainKey]),
+    closePopover,
+    disconnectAccount: (domainKey) => {
+      disconnectAccount(domainKey, selectedAddress)
+      if (connectedDomains.length === 1) {
+        closePopover()
```

Comment:
Hmm. I guess this seems reasonable! It hadn't occurred to me that we'd do this.

Reply:
It felt really weird to remove the last site and find myself looking at the empty popover.